src/components/program/ProgramSubscriptionPlanAdminCard.tsx

```diff
@@ -22,31 +24,65 @@ const messages = defineMessages({
 const StyledCountDownBlock = styled.div`
   margin-top: 20px;
 `
-const StyledMenuItemText = styled.span`
-  display: block;
-`
 
 const ProgramSubscriptionPlanAdminCard: React.FC<{
   programId: string
   programPlan: ProgramPlan
+  programPlans?: ProgramPlan[]
   onRefetch?: () => void
-}> = ({ programId, programPlan, onRefetch }) => {
+}> = ({ programId, programPlan, programPlans, onRefetch }) => {
   const { formatMessage } = useIntl()
   const { enabledModules } = useApp()
+  const theme = useAppTheme()
   const { salePrice, listPrice, discountDownPrice, periodType, periodAmount, currencyId } = programPlan
   const { loadingEnrollmentCount, enrollmentCount } = useProgramPlanEnrollmentCount(programPlan.id)
 
+  const [setProgramPlanPrimary] = useMutation<
+    hasura.UPDATE_PROGRAM_PLAN_PRIMARY,
+    hasura.UPDATE_PROGRAM_PLAN_PRIMARYVariables
+  >(UPDATE_PROGRAM_PLAN_PRIMARY)
+
+  const [archiveProgramPlan] = useMutation<hasura.DELETE_PROGRAM_PLAN, hasura.DELETE_PROGRAM_PLANVariables>(
+    DELETE_PROGRAM_PLAN,
+  )
+
   const isOnSale = (programPlan.soldAt?.getTime() || 0) > Date.now()
   const description = programPlan.description?.trim() || ''
   const programPlanType = programPlan.autoRenewed
     ? 'subscription'
     : programPlan.periodAmount && programPlan.periodType
     ? 'period'
     : 'perpetual'
+
+  const handleSetProgramPlanPrimary = () => {
+    setProgramPlanPrimary({ variables: { programId: programId, programPlanId: programPlan.id } }).then(() => {
+      message.success(formatMessage(commonMessages.event.successfullySaved))
+      onRefetch?.()
+    })
+  }
+  const handleArchiveProgramPlan = () => {
+    archiveProgramPlan({ variables: { programPlanId: programPlan.id } })
+      .then(() => {
+        onRefetch?.()
+      })
+      .then(() => {
+        if (programPlan.isPrimary && programPlans && programPlans.length > 1) {
+          setProgramPlanPrimary({
+            variables: {
+              programId: programId,
+              programPlanId: programPlans.filter(plan => plan.id !== programPlan.id)[0].id,
+            },
+          })
+          onRefetch?.()
+        }
+        message.success(formatMessage(commonMessages.event.successfullyDeleted))
+      })
```

Comment:
only need to refetch once after `archiveProgramPlan` and  `setProgramPlanPrimary`